pandas/core/indexes/datetimes.py

```diff
@@ -573,6 +590,9 @@ def intersection(self, other):
               not other.freq.isAnchored() or
               (not self.is_monotonic or not other.is_monotonic)):
             result = Index.intersection(self, other)
+            # Invalidate the freq of `result`, which may not be correct at
+            # this point, depending on the values.
+            result.freq = None
```

Comment:
is there a test specific to this?  If so I'll put this into the next "pieces of 24024 PR"

Reply:
No new tests, so I don't think this is easily separable. I don't recall the exact source of the differences, but I think it was the handling of `freq` in `_simple_new` or `_shallow_copy`, `_shallow_copy_with_infer` or the `__new__` / `__init__`.